test_runner/batch_others/test_proxy.py

```diff
@@ -23,6 +29,116 @@ def test_password_hack(static_proxy):
         static_proxy.safe_psql("select 1", sslsni=0, user=user, password=magic)
 
 
+def get_session_id_from_uri_line(uri_prefix, uri_line):
+    assert uri_prefix in uri_line
+
+    url_parts = urlparse(uri_line)
+    psql_session_id = url_parts.path[1:]
+    link_auth_uri_prefix = uri_line[: -len(url_parts.path)]
+    assert (
+        link_auth_uri_prefix == uri_prefix
+    ), f"Line='{uri_line}' should contain a http auth link of form '{uri_prefix}/<psql_session_id>'."
+    assert psql_session_id is not None, "did not find line containing " + uri_prefix
+
+    return psql_session_id
+
+
+def create_and_send_db_info(local_vanilla_pg, psql_session_id, mgmt_port):
+    pg_user = "proxy"
+    pg_password = "password"
+
+    local_vanilla_pg.start()
+    query = f"create user {pg_user} with login superuser password '{pg_password}'"
+    local_vanilla_pg.safe_psql(query)
+
+    port = local_vanilla_pg.default_options["port"]
+    host = local_vanilla_pg.default_options["host"]
+    dbname = local_vanilla_pg.default_options["dbname"]
+
+    db_info_dict = {
+        "session_id": psql_session_id,
+        "result": {
+            "Success": {
+                "host": host,
+                "port": port,
+                "dbname": dbname,
+                "user": pg_user,
+                "password": pg_password,
+            }
+        },
+    }
+    db_info_str = json.dumps(db_info_dict)
+    cmd_args = [
+        "psql",
+        "-h",
+        "127.0.0.1",  # localhost
+        "-p",
+        f"{mgmt_port}",
+        "-c",
+        db_info_str,
+    ]
+
+    log.info(f"Sending to proxy the user and db info: {' '.join(cmd_args)}")
+    p = subprocess.Popen(cmd_args, stdout=subprocess.PIPE)
+    out, err = p.communicate()
+    assert "ok" in str(out)
+
+
+async def get_uri_line_from_process_welcome_notice(link_auth_uri_prefix, proc):
+    max_num_lines_of_welcome_message = 3
+    for attempt in range(max_num_lines_of_welcome_message):
+        raw_line = await proc.stderr.readline()
+        line = raw_line.decode("utf-8").strip()
+        if link_auth_uri_prefix in line:
+            return line
+    assert False, "did not find line containing " + link_auth_uri_prefix
+
+
+async def read_process_bytes(proc):
+    return await proc.stdout.read()
```

Comment:
I don't get the value of this. We're replacing a well-documented stdlib function with an undocumented internal util. Which bytes are being read? Stderr or stdout? Can you tell from the name?

Reply:
because mypy complains about it, this is a hack to not deal with the the type miss-match that mypy reports.

Reply:
```
(venv) klimentserafimov@Kliments-MacBook-Pro neon % poetry run mypy test_runner/batch_others/test_proxy.py
test_runner/batch_others/test_proxy.py:133: error: Item "Tuple[object, ...]" of "Union[Tuple[int, str], Process]" has no attribute "stdout"
test_runner/batch_others/test_proxy.py:133: error: Item "None" of "Union[Any, StreamReader, None]" has no attribute "read"
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
ok found a way to fix it.